twill-core/src/main/java/org/apache/twill/internal/TwillRuntimeSpecification.java

```diff
@@ -105,4 +114,17 @@ public String getRmSchedulerAddr() {
   public Map<String, Integer> getMaxRetries() {
     return maxRetries;
   }
+
+  /**
+   * Returns the ZK connection string for the Kafka used for log collections,
+   * or {@code null} if log collection is disabled.
+   */
+  @Nullable
+  public String getKafkaZKConnect() {
+    if (!isLogCollectionEnabled()) {
```

Comment:
The call to `getZkConnectStr` and `getTwillAppName` and `getTwillAppRunId` will never return null based on the contract (no @Nullable) , hence call to `getKafkaZKConnect` will never be null. 

I was saying that either way, someone need to check whether call to getKafkaZKConnect will return null if log collection is disabled, so why not ask to isLogCollectionEnabled to check first.